rust/connlib/tunnel/src/device_channel/tun_android.rs

```diff
@@ -38,14 +37,7 @@ impl Tun {
         utils::poll_raw_fd(&self.fd, |fd| read(fd, buf), cx)
     }
 
-    pub fn new(
-        config: &InterfaceConfig,
-        dns_config: Vec<IpAddr>,
-        callbacks: &impl Callbacks,
-    ) -> Result<Self> {
-        let fd = callbacks
-            .on_set_interface_config(config.ipv4, config.ipv6, dns_config)
-            .ok_or(Error::NoFd)?;
+    pub fn with_fd(fd: RawFd) -> Result<Self> {
         // Safety: File descriptor is open.
```

Comment:
How is this function safe if its safety depends on the caller passing an open fd?

Reply:
It isn't but this is already what is on main, I just moved things around a bit.